### PR TITLE
Move xla tensor to cuda tensor (via CPU tensor)

### PR DESCRIPTION
**Pull request by vanbasten23**

### CODE REVIEW

File: test/test_operations.py [hunk 1]
```diff
@@ -2437,6 +2437,29 @@ def test_aten_move_scalar_cuda_to_xla(self):
...
+    print(met.metrics_report())
+
+  @onlyIfTorchSupportsCUDA
+  @onlyIfPJRTDeviceIsCUDA
+  def test_aten_move_scalar_xla_to_cuda(self):
+    # 0-dimensional scalar-tensor
+    # Has a different execution path than other tensors.
+    self._test_move_tensor_cuda_to_xla(torch.tensor(42))
+    print(met.metrics_report())
+
```
Comment (vanbasten23) [inline test/test_operations.py:2462]: note to myself: remove the metrics_report.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 2]
```diff
@@ -551,8 +551,8 @@ at::Tensor XLANativeFunctions::_to_copy(
 
   // Case 1: Materialize the tensor.
   if (device && device->type() != c10::kXLA) {
-    XLA_CHECK(device->type() == c10::kCPU)
-        << "only cpu device is supported in _to_copy.";
+    // XLA_CHECK(device->type() == c10::kCPU)
+    //     << "only cpu device is supported in _to_copy.";
     auto self_tensor = bridge::GetXlaTensor(self);
     auto eager_tensor = self_tensor->ToTensor(/*detached=*/true);
```
Comment (ysiraichi) [inline torch_xla/csrc/aten_xla_type.cpp:557]: Doesn't this code brings the tensor to CPU?

Comment (vanbasten23) [reply torch_xla/csrc/aten_xla_type.cpp:557]: Right. The eager_tensor after the line `auto eager_tensor = self_tensor->ToTensor(/*detached=*/true);` appear to be on CPU.

Do you need this PR to experiment the cuda->cpu->xla option? If not, I'll drop the PR and try to find a more direct way.

Comment (ysiraichi) [reply torch_xla/csrc/aten_xla_type.cpp:555]: Not really. Sorry for the late reply, by the way.

**Pull request closed without merging**